core/db/migrate/20150515211137_fix_adjustment_order_id.rb

```diff
@@ -0,0 +1,75 @@
+class FixAdjustmentOrderId < ActiveRecord::Migration
+  def change
+    say 'Populate order_id from adjustable_id where appropriate'
+    execute(<<-SQL.squish)
+      UPDATE
+        spree_adjustments
+      SET
+        order_id = adjustable_id
+      WHERE
+            adjustable_type = 'Spree::Order'
+        AND order_id IS NULL
+      ;
+
+      UPDATE
+        spree_adjustments
+      SET
+        order_id =
+          (SELECT order_id FROM spree_line_items WHERE spree_line_items.id = spree_adjustments.adjustable_id)
+      WHERE
+            adjustable_type = 'Spree::LineItem'
+        AND order_id IS NULL
+      ;
+
+      UPDATE
+        spree_adjustments
+      SET
+        order_id =
+          (SELECT order_id FROM spree_shipments WHERE spree_shipments.id = spree_adjustments.adjustable_id)
+      WHERE
+            adjustable_type = 'Spree::Shipment'
+        AND order_id IS NULL
+    SQL
```

Comment:
I believe the adapter doesn't split by `;` into separate statements. Which I think is semi-sensible.